srcpkgs/awesome/template

```diff
@@ -1,27 +1,40 @@
 # Template file for 'awesome'
 pkgname=awesome
 version=4.3
-revision=4
+revision=5
 build_style=cmake
 build_helper="qemu"
 configure_args="-DSYSCONFDIR=/etc"
 conf_files="/etc/xdg/awesome/rc.lua"
-hostmakedepends="ruby-asciidoctor ImageMagick lgi lua pkg-config xmlto"
+hostmakedepends="ruby-asciidoctor ImageMagick pkg-config xmlto"
 makedepends="libxcb-devel pango-devel xcb-util-devel xcb-util-image-devel
  xcb-util-keysyms-devel xcb-util-wm-devel xcb-util-cursor-devel
- startup-notification-devel imlib2-devel lgi libxdg-basedir-devel
- gdk-pixbuf-devel lua-devel dbus-devel libxkbcommon-devel xcb-util-xrm-devel"
-depends="dbus-x11 lgi>=0.7.2 pango"
+ startup-notification-devel imlib2-devel libxdg-basedir-devel
+ gdk-pixbuf-devel dbus-devel libxkbcommon-devel xcb-util-xrm-devel"
+depends="dbus-x11 pango"
 short_desc="Highly configurable, next gen framework window manager for X"
 maintainer="Orphaned <orphan@voidlinux.org>"
 license="GPL-2.0-or-later"
 homepage="http://awesomewm.org"
 distfiles="https://github.com/awesomeWM/awesome/releases/download/v${version}/awesome-${version}.tar.xz"
 checksum=78264d6f012350b371e339127aca485260bc0aa935eff578ba75ce1a00e11753
+build_options="jit"
+desc_option_jit="Use LuaJIT instead of lua"
+hostmakedepends+=" $(vopt_if jit lua51 lua)-lgi"
```

Comment:
all of these can be combined together so you don't have several `makedepends+=` etc, make sure to stick with 80 columns as well

Reply:
I got some problem while building awesome with `-o jit`

```
luajit: error loading module 'lgi.corelgilua51' from file '/usr/local/lib/lua/5.1/lgi/corelgilua51.so':
	/usr/local/lib/lua/5.1/lgi/corelgilua51.so: undefined symbol: lua_setuservalue
```

I think I need some more work on lgi. Any idea?

Reply:
that looks like it was compiled against incorrect includes

on linux, modules are not linked against lua, and `lua_setuservalue` didn't exist before 5.2

Reply:
> that looks like it was compiled against incorrect includes

So this doesn't have any effect at all? :confused:  :question: 

```
CFLAGS="$CFLAGS -I ${XBPS_CROSS_BASE}/usr/include/lua${luaver}"
```

Reply:
Oh my, I found out what the problem was and it is embarrassing. I actually polluted my `masterdir` with and old broken build because I didn't set the `PREFIX` in `do_install`, which causes the `awesome` build to fail every single time.